expressServer/index.js

```diff
@@ -12,55 +14,66 @@ const port = process.env.PORT || 5000;
 app.use(cors());
 app.use(express.json());
 
-const userSchema = new mongoose.Schema({
-  username: { type: String, required: true },
-  password: { type: String, required: true },
-});
-
-const User = mongoose.model('User', userSchema);
+const generateJWT = (user) => {
```

Comment:
You can move this to a lib. I suggest you to create a crypto.ts file and put all your 'hashPassword', 'saltUsername' and this method together there. 

I also think you should rename the function saltUsername to 'generateSalt' or something similar since it's more descriptive.

Reply:
Does it have to be a .ts?

Reply:
no sorry, in your case will be a js